test/e2e/create_cluster_test.go

```diff
@@ -232,5 +238,72 @@ func validatePrivateCluster(t *testing.T, ctx context.Context, client crclient.C
 		g.Expect(hostedCluster.Status.ControlPlaneEndpoint.Host).ToNot(ContainSubstring("hypershift.local"))
 	}
 
+	validateHostedClusterConditions(t, ctx, client, hostedCluster)
 	e2eutil.EnsureNoCrashingPods(t, ctx, client, hostedCluster)
 }
+
+func validateHostedClusterConditions(t *testing.T, ctx context.Context, client crclient.Client, hostedCluster *hyperv1.HostedCluster) {
+	expectedConditions := map[hyperv1.ConditionType]metav1.ConditionStatus{
```

Comment:
@muraee we also have this knowledge (about expected condition state) here: https://github.com/openshift/hypershift/blob/5d65434f8cb8627592c780b389ad23c724860480/hypershift-operator/metrics.go#L203 
It'd be great if we can consolidate this knowledge in a central place *and* if there is a condition set on the HC (or nodepool) that is not contained in our known list, fail this check.